core/src/main/scala/kafka/log/LogConfig.scala

```diff
@@ -69,6 +72,9 @@ case class LogConfig(props: java.util.Map[_, _]) extends AbstractConfig(LogConfi
   val minInSyncReplicas = getInt(LogConfig.MinInSyncReplicasProp)
   val compressionType = getString(LogConfig.CompressionTypeProp).toLowerCase
   val preallocate = getBoolean(LogConfig.PreAllocateEnableProp)
+  val messageFormatVersion = ApiVersion(getString(LogConfig.MessageFormatVersionProp))
```

Comment:
The thing is that the way TopicConfigCommand works is that it just writes the config in ZK and then writes the config notification. TopicConfigHandler just follows the notification. If we add the check in TopicConfigHandler, we need to remove the config in ZK, which makes things more complicated.

We can probably just leave this as a topic level config with the caveat that the config may be ignored if the broker property doesn't match.